BraintreeCore/src/main/java/com/braintreepayments/api/AuthorizationLoader.java

```diff
@@ -0,0 +1,50 @@
+package com.braintreepayments.api;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+class AuthorizationLoader {
+
+    private Authorization authorization;
+    private final ClientTokenProvider clientTokenProvider;
+
+    AuthorizationLoader(@Nullable String initialAuthString, @Nullable ClientTokenProvider clientTokenProvider) {
+        this.clientTokenProvider = clientTokenProvider;
+        if (initialAuthString != null) {
+            this.authorization = Authorization.fromString(initialAuthString);
+        }
+    }
+
+    void loadAuthorization(@NonNull final AuthorizationCallback callback) {
+        if (authorization != null) {
+            callback.onAuthorizationResult(authorization, null);
+        } else if (clientTokenProvider != null) {
+            clientTokenProvider.getClientToken(new ClientTokenCallback() {
+                @Override
+                public void onSuccess(@NonNull String clientToken) {
+                    authorization = Authorization.fromString(clientToken);
+                    callback.onAuthorizationResult(authorization, null);
+                }
+
+                @Override
+                public void onFailure(@NonNull Exception exception) {
+                    callback.onAuthorizationResult(null, exception);
+                }
+            });
+        } else {
+            callback.onAuthorizationResult(null, new BraintreeException("unable to fetch client token"));
```

Comment:
Nitpick and we should probably also callback a developer error if `ClientTokenProvider` is null
```suggestion
            callback.onAuthorizationResult(null, new BraintreeException("Unable to fetch client token"));
```